### PR TITLE
Use sklearn in runif

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

addresses: https://github.com/Lightning-AI/lightning/pull/15311#discussion_r1009251263

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: tests/legacy/simple_classif_training.py [hunk 1]
```diff
@@ -14,141 +14,16 @@
...
-from pytorch_lightning import LightningDataModule, LightningModule, seed_everything
+from pytorch_lightning import seed_everything
 from pytorch_lightning.callbacks import EarlyStopping
+from tests_pytorch.helpers.datamodules import ClassifDataModule
+from tests_pytorch.helpers.simple_models import ClassificationModel
 
 PATH_LEGACY = os.path.dirname(__file__)
 
 
-class SklearnDataset(Dataset):
```
Comment (rohitgr7) [inline tests/legacy/simple_classif_training.py:31]: this is duplicated in helpers

File: tests/tests_pytorch/helpers/datamodules.py [hunk 2]
```diff
@@ -54,7 +53,8 @@ def test_dataloader(self):
 class SklearnDataModule(LightningDataModule):
     def __init__(self, sklearn_dataset, x_type, y_type, batch_size: int = 10):
         if not _SKLEARN_AVAILABLE:
-            pytest.skip(str(_SKLEARN_AVAILABLE))
+            raise ImportError(str(_SKLEARN_AVAILABLE))
```
Comment (rohitgr7) [inline tests/tests_pytorch/helpers/datamodules.py:56]: now it raise an error if sklearn is not available

File: tests/tests_pytorch/accelerators/test_ipu.py [hunk 3]
```diff
@@ -149,7 +149,8 @@ def test_inference_only(tmpdir, devices):
     trainer.predict(model)
 
 
-@RunIf(ipu=True)
+@RunIf(ipu=True, sklearn=True)
+@mock.patch.dict(os.environ, os.environ.copy())
```
Comment (rohitgr7) [inline tests/tests_pytorch/accelerators/test_ipu.py:153]: some env_vars were leaking

**Review by awaelchli — APPROVED** [review]

File: requirements/pytorch/test.txt [hunk 4]
```diff
@@ -8,7 +8,6 @@ pre-commit==2.20.0
 
 # needed in tests
 cloudpickle>=1.3, <=2.1.0
-scikit-learn>0.22.1, <1.1.3
```
[the remark this answers is not in the document]
Comment (rohitgr7) [reply requirements/pytorch/test.txt]: this is outdated.. it is kept as it is
